assets/javascripts/discourse/components/chat-composer.js

```diff
@@ -680,6 +680,24 @@ export default Component.extend(TextareaTextManipulation, {
     this.onValueChange?.(this.value, this._uploads, this.replyToMsg);
   },
 
+  @action
+  onTextareaFocusin(target) {
+    if (!this.capabilities.isIOS) {
+      return;
+    }
+
+    // hack to prevent the whole viewport
+    // to move on focus input
+    target = document.querySelector(".chat-composer-input");
+    target.style.transform = "translateY(-99999px)";
+    target.focus();
+    window.requestAnimationFrame(() => {
```

Comment:
Double RAF? 😱 

Can you add a comment to explain why this is required/how it works?

Reply:
there's already the word hack at the top of the block, what else do you want ? :D

Reply:
I guess if there's an explanation of how the hacks works, it gets easier to know when to delete it.

Reply:
AFAIK this is because of this: https://stackoverflow.com/a/44146560

I'm not really sure I want to carry this comment, if you remove this, you should know what you do.